php-frontend/src/main/java/org/sonar/plugins/php/api/tree/declaration/MethodDeclarationTree.java

```diff
@@ -74,4 +75,10 @@ public interface MethodDeclarationTree extends ClassMemberTree, FunctionTree {
   @Override
   Tree body();
 
+  /**
+   * Provide access to the symbol associated to the method.
+   * @return the symbol of the method, null if the symbol has not been resolved
+   */
+  @Nullable
+  MethodSymbol symbol();
```

Comment:
`MethodSymbol` should not be mentioned in the interface, main because this class is part of the `api`.
In the implementing class we have the interface `HasMethodSymbol` which is used to indicate a symbol.

Reply:
Tbh, I think I would prefer the cast solution. I'm not 100% familiar with the restrictions of the API, but the less we change the better it is for us.
In the future we will get more experience with the go frontend, so maybe I will change my opinion in the future.

Reply:
Otherwise we need to ask the security squad (maybe karim / nils would be the best points of contact but they're both on holiday), what solution they would prefer.

Reply:
I understand, let's keep this then, at least for now.